src/main/java/software/amazon/event/kafkaconnector/batch/EventBridgeBatchingStrategy.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector.batch;
+
+import java.util.List;
+import java.util.stream.Stream;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequestEntry;
+import software.amazon.event.kafkaconnector.util.MappedSinkRecord;
+
+@FunctionalInterface
+public interface EventBridgeBatchingStrategy {
+
+  Stream<List<MappedSinkRecord<PutEventsRequestEntry>>> apply(
```

Comment:
Great abstraction! nit: Can you elaborate why we work with Stream instead of List here? Afaik we will already get a List MappedSinkRecord<PutEventsRequestEntry> as input param and the result would ideally be a List of List already so the caller does not have to .stream and .collect again?

Reply:
The result of batching is streamed to `sendToEventBridge`, the type of each stream item is the batched list of `PutEventsRequestEntry`:
```
var sendItemResults =
    batching
        .apply(mappingResult.success.stream())
        .flatMap(this::sendToEventBridge)
        .collect(toList());
```
The input can be a list instead of a stream, so the snippet would be changed to:
```
var sendItemResults =
    batching
        .apply(mappingResult.success)
        .flatMap(this::sendToEventBridge)
        .collect(toList());
```

Reply:
@embano1, @maschnetwork I've started to play with the 'Best Effort' batching strategy (#26). I forgot to take errors in to account on the strategy interface. At the moment each `SinkRecord` is send to Event Bridge even if it's to large. Should we do this also in the 'Best Effort' batching strategy or should we change the interface to return proper batches and obviously failing items as `*Result`?

Reply:
Good catch. I see three potential ways to handle this:

1. Ignore and don't let Batching deal with errors. 
    **Pros**: Simplicity **Cons**: Additional API Call for already known error
2. Implement another wrapper class similar to 'EventBridgeMappingResult' but with a List of Lists for the success for batching 
   **Pros**: No API Calls for known error **Cons**: Additional Classes and Error Handling in the Writer 
3. Make Batching a responsbility of the Mapper and return EventBridgeMappingResult but with a List of Lists for the success. 
   **Pros**: One place to deal with errors (pre-API) and reduce number of needed Wrapper Classes **Cons**: Batching hidden in mapper 

@embano1 wdyt?

Reply:
@agebhar1 good catch! @maschnetwork great suggestions, makes it easier for me :-p

Tbh, I'd go with option 1 for now as we don't really know about API errors and currently defer any validation to either Kafka Connect (during deserialization of incoming records before passing to our sink implementation) or EventBridge service API.

Reply:
I was playing with
```
@FunctionalInterface
public interface EventBridgeBatchingStrategy {

  Stream<List<EventBridgeResult<PutEventsRequestEntry>>> apply(
      Stream<MappedSinkRecord<PutEventsRequestEntry>> records);
}
```
which should work. We can improve this in future.

Reply:
> which should work. We can improve this in future.

Can you elaborate a bit on "which should work"? Not sure I'm 100% following

Reply:
If we change the signature of the batching strategy to:
```
@FunctionalInterface
public interface EventBridgeBatchingStrategy {

  Stream<List<EventBridgeResult<PutEventsRequestEntry>>> apply(
      Stream<MappedSinkRecord<PutEventsRequestEntry>> records);
}
```
there is only a little effort to update the `EventBridgeWriter` to the interface. Each failed `PutEventsRequestEntry` with it's corresponding `SinkRecord` is then returned by the `EventBridgeWriter` and can be handled in the `EventBridgeSinkTask`.

Reply:
@embano1, @maschnetwork I've add this 38fc5dc821d81cedc3711d82c47127503035a314 commit to address the error on batching.

Reply:
@agebhar1 I checked the commit and while I understand the reasoning I would go with @embano1 proposed approach for now and accept the current single record > 256kb to be handeled by the SDK error handling. I think at this point making EventBridgeResult generic creates additional overhead on the method signatures and reduces readability. We can also create a tracking issue for this and handle that separately so we can get the current work merged first and have that as a seperate discussion.

Reply:
+1 on readability, let's revert then?